clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,506 @@
+	}
+
+	if err := npm.initializeInformer(); err != nil {
+		return err
+	}
+
+	if err := npm.executeTest(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) executeTest() error {
+	switch npm.podRatioType {
+	case oneToOne, manyToMany:
+		npm.execNToMTest()
+	default:
+		return fmt.Errorf("invalid Pod Ratio")
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) initialize(config *measurement.Config) {
+	npm.k8sClient = config.ClusterFramework.GetClientSets().GetClient()
+	npm.framework = config.ClusterFramework
+	npm.dynamicClient = config.ClusterFramework.GetDynamicClients().GetClient()
+	npm.podInfo.workerPodMap = make(map[string]podList)
+	npm.podInfo.podPairCh = make(chan uniquePodPair)
+}
+
+func (npm *networkPerformanceMeasurement) initializeInformer() error {
+	gvr, _ := meta.UnsafeGuessKindToResource(crGvk)
+	npm.resourceInterface = npm.dynamicClient.Resource(gvr).Namespace(netperfNamespace)
+	informer, err := getInformer(netperfNamespace, npm.dynamicClient, gvr)
+	if err != nil {
+		return fmt.Errorf("error getting informer:%s", err)
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+
+	// Create clusterRoleBinding
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating clusterRoleBinding: %v", err)
+	}
+
+	// Create CRD
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	// Delete CRD
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	// Delete clusterRoleBinding
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	npm.templateMapping = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
```

Comment:
overwriting np.templateMapping if you only set `Replicas` here doesn't sound like the best option.

Please create a dedicate variable here  and use it.

Reply:
Done